prune.go

```diff
@@ -27,19 +27,36 @@ const (
 	// logInterval defines often to report progress during long
 	// operations.
 	logInterval = 15 * time.Second
+
+	// maxIterCount is the number of times we will pass over the data to make sure all documents are cleaned up.
+	// (removing from a collection you are iterating can cause you to miss entries). The loop should exit early if
+	// it finds nothing to do anyway, so this only affects the number of times we will evaluate documents we aren't
+	// removing.
+	maxIterCount = 5
+
+	// maxMemoryTokens caps our in-memory cache. When it is full, we will apply our current list of items to process, and then flag the loop to run again.
+	maxMemoryTokens = 100000
```

Comment:
You mention that this results in a max memory usage of 1GB. I'm a little worried that 1GB might be a bit much on some controller machines. Have you played with smaller values to see how it affects performance?